### ISSUE TITLE
Amazfit Bip - This activity does not contain GPX tracks

### ISSUE THREAD

**Issue by MortezaAghazamani** [Freeyourgadget/Gadgetbridge#1647]
#### Before reporting a bug, please confirm the following:
- [x] I have read the [wiki](https://github.com/Freeyourgadget/Gadgetbridge/wiki), and I didn't find a solution to my problem / an answer to my question.
- [x] I have searched the [issues](https://github.com/Freeyourgadget/Gadgetbridge/issues), and I didn't find a solution to my problem / an answer to my question.
- [x] If you upload an image or other content, please make sure you have read and understood the [github policies and terms of services](https://help.github.com/articles/github-terms-of-service/#1-responsibility-for-user-generated-content)

### I got Gadgetbridge from:
* [x] F-Droid
* [ ] I built it myself from source code (specify tag / commit)

#### My issue is:
When I click on fetched activities from Activities tab. I see an error **"This activity does not contain GPX tracks."**.

[gadgetbridge.log](https://github.com/Freeyourgadget/Gadgetbridge/files/3628026/gadgetbridge.log)

#### My wearable device is:
Amazfit Bip
FW: 1.1.5.56
GPS: 18344,eb2f43f,126

#### My android version is:
9

#### My Gadgetbridge version is:
HW: V0.11.19.5

**Comment by vanous** [comment]
This is most probably "thread mill" activity. These are currently not utilized in GB. If you use activities with GPS (Running...), these will be exported as GPX file.

This is actually documented here: https://codeberg.org/Freeyourgadget/Gadgetbridge/wiki/Amazfit-Bip-export-of-activities

**Comment by MortezaAghazamani** [comment]
**Thank you for your reply.**
Last day I found something new with second activity.
When I recorded my first activity it was an outdoor running activity. I fetched that activity, but the folder “android/data/nodomain.freyourgadget.gadgetbridge/files” was empty with no GPX file. Also when I clicked on that activity, I saw that message I posted above.

![Screenshot_2019-09-21-00-28-47-181_nodomain freeyourgadget gadgetbridge](https://user-images.githubusercontent.com/20865770/65355938-6858a600-dc08-11e9-9d8c-2028e1a711dd.png)

But when I started the **second activity**, a file was created there. That file contained my second activity path but with a wrong start position. It was started from my first activity position.
In addition, I tested my recorded activities on other App and I was able to watch them correctly on the map.
What could I do for solve this issue? More information or more test?

**Comment by vanous** [comment]
GPS takes some time to get a fix. For me, this works really well, I use it about once in ten days, so not very often.

**Comment by MortezaAghazamani** [comment]
Yes, it takes time to decode and synchronize with the GPS signal. In fact, that file contains GPS tracks and I can see it correctly inside the watch also in another app. But I can't export it correctly from the Gadgetbridge. maybe cause of some incompatibility with the format of the data. Or version incompatibility. 🙄😞

**Comment by vanous** [comment]
What do you mean by another app? You can only observe GPX in another app :smile:

**Comment by MortezaAghazamani** [comment]
No no 😄 , That's not what I meant. I tested my recorded activity in other APPS like "Notify & Fitness for Amazfit" and it works well at that APP. I'm sure about my watch works well. But "Gadgetbridge" at this version does not working well with my watch.

**Comment by ashimokawa** [comment]
So the Problem seems to be a mismatch between recorded activity and activity summary.

**Comment by MortezaAghazamani** [comment]
Activity fetch does not work on Amazfit Bip. Gadgetbridge always fetch wrong data from my watch. It seems like there is no plan to solve this issue. I'm totally disappointed. Unfortunately I'm not an Android developer. ‌By the way, is there anyone to guide me? And tell me where should I start to fix this bug?

**Comment by vanous** [comment]
> Activity fetch does not work on Amazfit Bip. Gadgetbridge always fetch wrong data from my watch. It seems like there is no plan to solve this issue. I'm totally disappointed. Unfortunately I'm not an Android developer. ‌By the way, is there anyone to guide me? And tell me where should I start to fix this bug?

Use our wiki. Make sure you can compile this project in Android Studio. Then, take the error message you provided here: _This activity does not contain GPX tracks._ , use the search everywhere feature (Ctrl+Shift+F) and search for that text. One of the places this will take you to is `ActivitySummariesActivity`. From there, you will easily get to `AmazfitBipSupport`and perhaps even to `FetchSportsSummaryOperation` → `FetchActivityOperation` ... and you can go on and on. After some time, you might even get to `HuamiActivityDetailsParser` and other interesting places. That is as much as me, non Android developer, can tell you. Hope this helps.
